src/cross_toml.rs

```diff
@@ -26,14 +27,41 @@ pub struct CrossBuildConfig {
 
 /// Target configuration
 #[derive(Debug, Deserialize, PartialEq, Eq)]
+#[serde(rename_all = "kebab-case")]
 pub struct CrossTargetConfig {
     xargo: Option<bool>,
     image: Option<String>,
+    #[serde(default, deserialize_with = "opt_string_or_struct")]
+    dockerfile: Option<CrossTargetDockerfileConfig>,
+    pre_build: Vec<String>,
     runner: Option<String>,
     #[serde(default)]
     env: CrossEnvConfig,
 }
 
+/// Dockerfile configuration
+#[derive(Debug, Deserialize, PartialEq, Eq)]
+#[serde(rename_all = "kebab-case")]
+pub struct CrossTargetDockerfileConfig {
+    file: String,
+    provide_base: bool,
+    context: Option<String>,
+    build_args: Option<HashMap<String, String>>,
```

Comment:
should this be a hashmap?

Reply:
I think it's fine. I'd probably have used a `Vec` since we never actually index the values, but the deduplication is nice. I might have used a `BTreeMap` instead in that case though.